src/api/app/models/flag.rb

```diff
@@ -38,16 +38,12 @@ def arch
   end
 
   def default_status
-    all_flag = main_object.flags.find_by('flag = ? AND repo IS NULL AND architecture_id IS NULL', flag)
-    repo_flag = main_object.flags.find_by('flag = ? AND repo = ? AND architecture_id IS NULL', flag, repo)
-    arch_flag = main_object.flags.find_by('flag = ? AND repo IS NULL AND architecture_id = ?', flag, architecture_id)
-    same_flag = main_object.flags.find_by('flag = ? AND repo = ? AND architecture_id = ?', flag, repo, architecture_id)
-    if main_object.is_a?(Package)
-      same_flag ||= main_object.project.flags.find_by('flag = ? AND repo = ? AND architecture_id = ?', flag, repo, architecture_id)
-      repo_flag ||= main_object.project.flags.find_by('flag = ? AND repo = ? AND architecture_id IS NULL', flag, repo)
-      arch_flag ||= main_object.project.flags.find_by('flag = ? AND repo IS NULL AND architecture_id = ?', flag, architecture_id)
-      all_flag ||= main_object.project.flags.find_by('flag = ? AND repo IS NULL AND architecture_id IS NULL', flag)
-    end
+    pkg_or_prj_flags = main_object.is_a?(Package) ? main_object.project.flags : main_object.flags
```

Comment:
careful. this is not the equivalent code. repo_flag for packag is package.flag || project.flag - you set it to package.flag

Reply:
but the good news: this function is not used anymore - if you cleaned up the unit tests

Reply:
Was still used by Specified::Flags rspec test, but I adapted the test, so the default_status method is gone now :)